RZBluetooth/RZBPeripheral.h

```diff
@@ -0,0 +1,184 @@
+//
+//  RZBPeripheral.h
+//  RZBluetooth
+//
+//  Created by Brian King on 3/22/16.
+//  Copyright © 2016 Raizlabs. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "RZBDefines.h"
+
+@class RZBCentralManager;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RZBPeripheral : NSObject
+
+/**
+ *  Designated initializer to over-ride by subclasses. You should not invoke this method directly.
+ */
+- (instancetype)initWithCorePeripheral:(CBPeripheral *)corePeripheral
+                        centralManager:(RZBCentralManager *)centralManager NS_DESIGNATED_INITIALIZER;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ *  The identifier of the backing Core Bluetooth peripheral.
+ */
+@property (strong, nonatomic, readonly) NSUUID *identifier;
+
+/**
+ *  The name of the backing Core Bluetooth peripheral
+ */
+@property (retain, readonly, nullable) NSString *name;
+
+/**
+ *  The state of the backing Core Bluetooth peripheral.
+ */
+@property (nonatomic, readonly) CBPeripheralState state;
+
+/**
+ *  The service objects of the backing Core Bluetooth peripheral.
+ */
+@property (retain, readonly, nullable) NSArray<CBService *> *services;
+
+/**
+ * The block to execute on connection
+ */
+@property (copy, nonatomic, nullable) RZBErrorBlock onConnection;
```

Comment:
For `RZBPeripheral` we need to decide what kind of callbacks we want. It's really confusing having:
```objc
@property (copy, nonatomic, nullable) RZBErrorBlock onConnection;
@property (copy, nonatomic, nullable) RZBErrorBlock onDisconnection;
```
and 
```
- (void)cancelConnectionWithCompletion:(RZBErrorBlock __nullable)completion;
- (void)connectWithCompletion:(RZBErrorBlock __nullable)completi
```
I know in the comments it says it won't trigger, but that is places you need duplicate code. IMO i think we need either 1 block or 1 delegate callback that will do it on connection status change (connecting and disconnecting) that will get called whether or not it is cancelled. 

So it could just be 
```objc
- (void)connectWithStatusChangeBlock:(RZBErrorBlock __nullable)onStatusChange;
- (void)cancelConnection; // this will trigger the statusChangeBlock
```

And it will be obvious what happens. You can obviously substitute  the block for a delegate callback, but the block was easier to write quickly